daemon/src/xtra_ext.rs

```diff
@@ -1,5 +1,7 @@
 use async_trait::async_trait;
+use futures_timer::Delay;
```

Comment:
It would be better to use the `tokio` primitive here, otherwise we have two timer reactors running.

We already depend on tokio, might as well use `tokio::time::sleep`!

Reply:
I thought that at some point we might try to upstream our changes - in that case `tokio` dependency might be a bit harder, but I guess if that ever happens it would trivial to change then. Removed the `futures_timer` dependency and used `tokio::time::sleep`